N24DECE089/buoi4/cau8.cpp

```diff
@@ -0,0 +1,14 @@
+#include <iostream> 
+#include <cmath> 
+using namespace std  ; 
+int main() 
+{
+   double a , b , c ;
+   cout << " moi nhap ba canh a , b , c cua tam giac  " ; cin >> a >> b >> c ;
+   if ( a == b || b == c || a == c ) { cout << " day la tam giac can " ; }
+   else if ( a == b == c ) { cout << " day la tam giac deu "; }
```

Comment:
1. biểu thức sẽ có sự tham gia của 2 toán hạng và 1 toán tử
nếu em viết như này máy nó sẽ hiểu là ( (a == b) == c )
(a == b) thì nó sẽ trả về true/false tương ứng là 1/0
vậy lúc này biểu thức sẽ trở thành ( 1 == c) hoặc (0 == c) vậy  còn đúng nữa ko?
cách viết của em là viết trên giấy, khi thể hiện lên code sẽ phải ghi rõ ràng ra: ` (a == b) && (b == c) `

2. phải kiểm tra tam giác đều trước tam giác cân
giờ anh nhập 3 3 3 thì đụng cái điều kiện tam giác đều trước và nó thoả mãn => kết luận tam giác cân, nhưng đúng ra lại là tam giác đều